common/output-model.cpp

```diff
@@ -82,32 +94,39 @@ void output_model::thread_loop()
                                 }
                             }
 
-                            if (!parsed_ok)
+                            if( ! parsed_ok )
                             {
                                 std::stringstream ss;
-                                for (auto& elem : message.data())
-                                    ss << std::setfill('0') << std::setw(2) << std::hex << static_cast<int>(elem) << " ";
-                                add_log(message.get_severity(), __FILE__, 0, ss.str());
+                                for( auto & elem : message.data() )
+                                    ss << std::setfill( '0' ) << std::setw( 2 ) << std::hex
+                                       << static_cast< int >( elem ) << " ";
+                                add_log( message.get_severity(), __FILE__, 0, ss.str() );
                             }
-                            if( ! enable_firmware_logs && fwlogger.get_number_of_fw_logs() == 0 )
-                            {
-                                fwlogger.stop_collecting();
+
+                            if( !enable_firmware_logs && fwlogger.get_number_of_fw_logs() == 0 )
                                 break;
-                            }
                         }
                     }
+                    catch( const std::exception & ex )
+                    {
+                        add_log( RS2_LOG_SEVERITY_WARN,
+                                 __FILE__,
+                                 __LINE__,
+                                 rsutils::string::from() << "Failed to fetch firmware logs: " << ex.what() );
+                    }
                 }
-                catch(const std::exception& ex)
-                {
-                    add_log( RS2_LOG_SEVERITY_WARN,
-                             __FILE__,
-                             __LINE__,
-                             rsutils::string::from() << "Failed to fetch firmware logs: " << ex.what() );
-                }
+                                    
+                // FW define the logs polling intervals to be no less than 100msec to cope with limited resources.
+                // At the same time 100 msec should guarantee no log drops
+                std::this_thread::sleep_for( std::chrono::milliseconds( 100 ) );
             }
-        // FW define the logs polling intervals to be no less than 100msec to cope with limited resources.
-        // At the same time 100 msec should guarantee no log drops
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+
+            // User dectivated FW logs. Stop collecting logs.
+            for( auto & fwlogger : loggers )
+                fwlogger.first.stop_collecting();
+        }
+
+        std::this_thread::sleep_for( std::chrono::milliseconds( 100 ) );
```

Comment:
Why do need to sleep here?

Reply:
The sleep here is for times when `enable_firmware_logs` is false, so the thread won't use much resources.